### PR TITLE
Added invokeApply parameter to $http to skip apply

### PR DESCRIPTION
**Pull request by Utsav2**
There are a bunch of times when we don't want a $rootScope.$apply after an http request succeeds. [This](http://stackoverflow.com/questions/21935215/make-angularjs-skip-running-a-digest-loop-if-http-get-resulted-in-no-new-data) stackoverflow question gives a good use case: polling.
We can also use this to do scope specific digests, which really helps performance and snappiness on mobile devices.

It also gives us a little more control over the API, and also makes it similar to $timeout.

### CODE REVIEW

**Comment by Utsav2** [comment]
I signed it!

File: src/ng/http.js [hunk 1]
```diff
@@ -1276,11 +1284,15 @@ function $HttpProvider() {
...
         }
 
-        if (useApplyAsync) {
-          $rootScope.$applyAsync(resolveHttpPromise);
-        } else {
-          resolveHttpPromise();
-          if (!$rootScope.$$phase) $rootScope.$apply();
+        var skipApply = (isDefined(invokeApply) && !invokeApply);
+
+        if (!skipApply) {
```
Comment (lgalfaso) [inline src/ng/http.js]: so if `skipApply` is true, then `resolveHttpPromise` will not get called?

Comment (Utsav2) [inline src/ng/http.js]: Yeah, that's a mistake.

**Comment by Utsav2** [comment]
Could you have a look at this #12557? I've fixed the problem.

**Pull request closed without merging**